### PR TITLE
update ceph-csi rbd snapshot document

### PR DESCRIPTION
**Pull request by Madhu-1**
**Description of your changes:**

The current document is too old and it has a reference to the master branch(earlier master branch referred to csi-v0.3.0).
This PR updated the snapshot document wrt to the changes required for csi-v1.0.0

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

[skip ci]

### CODE REVIEW

**Comment by Madhu-1** [comment]
@travisn @rootfs PTAL

**Comment by galexrt** [comment]
@Madhu-1 Please add `[skip ci]` for documentation only changes so the CI won't run for it. I have added it to the PR now.

**Comment by Madhu-1** [comment]
@galexrt Thanks forgot to add it

File: Documentation/ceph-csi-drivers.md [hunk 1]
```diff
@@ -295,6 +317,42 @@ SNAPID NAME
      4 csi-rbd-pvc-c20495c0d5de11e8-snap-4c0b455b-d5fe-11e8-bebb-525400123456 1024 MB Mon Oct 22 13:28:03 2018
 ```
 
+#### Restore the snapshot to a new PVC
+
+```yaml
```
Comment (travisn) [inline Documentation/ceph-csi-drivers.md]: could we include these yamls in the example folder? Then users don't actually have to copy-paste.

**Comment by Madhu-1** [comment]
@travisn updated PR, PTAL

File: Documentation/ceph-csi-drivers.md [hunk 2]
```diff
@@ -117,49 +151,24 @@ kubectl exec -ti -n rook-ceph rook-ceph-mon-a-6c4f9f6b6-rzp6r -- bash -c "ceph -
 
 Then create a Secret using admin and `kubernetes` keyrings:
 
-## rbd-secret.yaml
+In [secret](../cluster/examples/kubernetes/ceph/csi/example/rbd/secret.yaml), you
+need your Ceph admin/user password encoded in base64. Run `ceph auth ls` in one
+of your Ceph pods, encode the key of your admin/user and replace
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]: Perhaps we can be more precise, e.g:

"to encode your key, run `echo KEY | base64`."

File: Documentation/ceph-csi-drivers.md [hunk 3]
```diff
@@ -252,30 +214,26 @@ In short, as the documentation describes it:
 > VolumeSnapshotClass provides a way to describe the “classes” of storage when
 > provisioning a volume snapshot.
 
-You must download this file and modify it to match your Ceph cluster.
+In [snapshotClass](cluster/examples/kubernetes/ceph/csi/example/rbd/snapshotclass.yaml),
+the `csi.storage.k8s.io/snapshotter-secret-name` parameter should reference the
+name of the secret created for the rbdplugin. The monitors are a comma
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
name of the secret created for the rbdplugin. The `monitors` are a comma
```

File: Documentation/ceph-csi-drivers.md [hunk 4]
```diff
@@ -206,44 +198,14 @@ pvc-c20495c0d5de11e8
...
-deployment, you must edit this file to match your configuration.
-
-```console
-kubectl create -f  https://raw.githubusercontent.com/ceph/ceph-csi/master/examples/rbd/csi-snapshotter.yaml
-```
-
-### Test RBD csi-snapshotter
-
-Next you need to create the SnapshotClass. The purpose of a SnapshotClass is
+You need to create the SnapshotClass. The purpose of a SnapshotClass is
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
You need to create the `SnapshotClass`. The purpose of a `SnapshotClass` is
```

File: Documentation/ceph-csi-drivers.md [hunk 5]
```diff
@@ -252,30 +214,26 @@ In short, as the documentation describes it:
...
-```console
-kubectl create -f snapshotclass.yaml
-```
+#### create volumesnapshot
 
-Finally, create the VolumeSnapshot resource. its `snapshotClassName` should be
-the name of the VolumeSnapshotClass previously created. The source name should
-be the name of the PVC you created earlier.
+In [snapshot](cluster/examples/kubernetes/ceph/csi/example/rbd/snapshot.yaml),
+`snapshotClassName` should be the name of the VolumeSnapshotClass previously
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
`snapshotClassName` should be the name of the `VolumeSnapshotClass` previously
```

File: Documentation/ceph-csi-drivers.md [hunk 6]
```diff
@@ -295,24 +253,46 @@ SNAPID NAME
      4 csi-rbd-pvc-c20495c0d5de11e8-snap-4c0b455b-d5fe-11e8-bebb-525400123456 1024 MB Mon Oct 22 13:28:03 2018
 ```
 
+#### Restore the snapshot to a new PVC
+
+In
+[pvc-restore](cluster/examples/kubernetes/ceph/csi/example/rbd/pvc-restore.yaml),
+`dataSource` should be the name of the VolumeSnapshot previously
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
`dataSource` should be the name of the `VolumeSnapshot` previously
```

File: Documentation/ceph-csi-drivers.md [hunk 7]
```diff
@@ -295,24 +253,46 @@ SNAPID NAME
...
+created. The kind should be the `VolumeSnapshot`.
+
+create a new PVC from the snapshot
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
Create a new PVC from the snapshot
```

File: Documentation/ceph-csi-drivers.md [hunk 8]
```diff
@@ -295,24 +253,46 @@ SNAPID NAME
...
+NAME              STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+rbd-pvc           Bound    pvc-84294e34-577a-11e9-b34f-525400581048   1Gi        RWO            csi-rbd        34m
+rbd-pvc-restore   Bound    pvc-575537bf-577f-11e9-b34f-525400581048   1Gi        RWO            csi-rbd        8s
+```
+
 ## RBD resource Cleanup
 
 To clean your cluster of the resources created by this example, run the following:
 
+if you have tested snapshot, delete snapshotclass,snapshot and pvc-restore
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
if you have tested snapshot, delete snapshotclass, snapshot and pvc-restore
```

File: Documentation/ceph-csi-drivers.md [hunk 9]
```diff
@@ -327,54 +307,34 @@ CRD](https://github.com/rook/rook/blob/master/Documentation/ceph-filesystem-crd.
...
+kubectl create -f cluster/examples/kubernetes/ceph/csi/example/cephfs/storegeclass.yaml
+```
+
 ## Create CephFS Secret
 
 Create a Secret that matches `provisionVolume` type specified in the [storageclass](../cluster/examples/kubernetes/ceph/csi/example/cephfs/storageclass.yaml).
 
-### cephfs-secret.yaml
+In [secret](../cluster/examples/kubernetes/ceph/csi/example/cephfs/secret.yaml) you
+need your Ceph admin/user ID and password encoded in base64.  Encode admin/user
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
need your Ceph admin/user ID and password encoded in base64. Encode admin/user
```

File: Documentation/ceph-csi-drivers.md [hunk 10]
```diff
@@ -327,54 +307,34 @@ CRD](https://github.com/rook/rook/blob/master/Documentation/ceph-filesystem-crd.
...
+ID in base64 format and replace `BASE64-ENCODED-USER` and Run `ceph auth ls` in
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
ID in base64 format and replace `BASE64-ENCODED-USER` and run `ceph auth ls` in
```

File: Documentation/ceph-csi-drivers.md [hunk 11]
```diff
@@ -327,54 +307,34 @@ CRD](https://github.com/rook/rook/blob/master/Documentation/ceph-filesystem-crd.
...
-Here, you need your Ceph admin/user ID and password encoded in base64.  Encode
-admin/user ID in base64 format and replace `BASE64-ENCODED-USER` and Run `ceph
-auth ls` in one of your Ceph pods, encode the key of your admin/user and replace
-`BASE64-ENCODED-PASSWORD` by your encoded key.
-
 ## Create CephFS PersistentVolumeClaim
 
-### cephfs-pvc.yaml
+In [pvc](../cluster/examples/kubernetes/ceph/csi/example/cephfs/pvc.yaml),
+make sure your `storageClassName` is the name of the StorageClass previously
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
make sure your `storageClassName` is the name of the `StorageClass` previously
```

File: Documentation/ceph-csi-drivers.md [hunk 12]
```diff
@@ -327,54 +307,34 @@ CRD](https://github.com/rook/rook/blob/master/Documentation/ceph-filesystem-crd.
...
+defined in storageclass.yaml
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
defined in `storageclass.yaml`
```

File: Documentation/ceph-csi-drivers.md [hunk 13]
```diff
@@ -389,35 +349,20 @@ preventing the PVC from being up and bound.
...
-  volumes:
-   - name: mypvc
-     persistentVolumeClaim:
-       claimName: cephfs-pvc
-       readOnly: false
+```console
+kubectl create -f cluster/examples/kubernetes/ceph/csi/example/cephfs/pod.yaml
 ```
 
+once the PVC is attached to the pod, pod creation process will continue
```
Comment (leseb) [inline Documentation/ceph-csi-drivers.md]:
```suggestion
Once the PVC is attached to the pod, the pod creation process will continue.
```

**Comment by Madhu-1** [comment]
@leseb Thanks for the review, done requested changes PTAL

File: Documentation/ceph-csi-drivers.md [hunk 14]
```diff
@@ -117,49 +151,24 @@ kubectl exec -ti -n rook-ceph rook-ceph-mon-a-6c4f9f6b6-rzp6r -- bash -c "ceph -
 
 Then create a Secret using admin and `kubernetes` keyrings:
 
-## rbd-secret.yaml
+In [secret](../cluster/examples/kubernetes/ceph/csi/example/rbd/secret.yaml), you
+need your Ceph admin/user password encoded in base64. Run `ceph auth ls` in one
+of your Ceph pods, to encode the key of your admin/user run `echo KEY |base64`
```
Comment (ktdreyer) [inline Documentation/ceph-csi-drivers.md]: This is a run-on sentence. Mind breaking it up?

Also, would you please provide a complete example command (kubectl exec?) so users know exactly where and how run `ceph auth ls`?

Comment (Madhu-1) [reply Documentation/ceph-csi-drivers.md]: @travisn @leseb am trying to create a  new user as  @rootfs suggested [here](https://github.com/rook/rook/pull/2934/files#diff-8be52e0c33719742b3b5ee58f5ae6f74L115)
is this command still valid or am i missing something?
```
$ ./kubectl -nrook-ceph exec -it rook-ceph-mon-a-67f5cb85c8-dgpjg -- bash -c "ceph -c /var/lib/rook/rook-ceph/rook-ceph.config auth get-or-create-key client.kubernetes mon \"allow profile rbd\" osd \"profile rbd pool=rbd\""
2019-04-09 07:23:43.039 7f974ce26700 -1 Errors while parsing config file!
2019-04-09 07:23:43.039 7f974ce26700 -1 parse_file: cannot open /var/lib/rook/rook-ceph/rook-ceph.config: (2) No such file or directory
Error initializing cluster client: ObjectNotFound('error calling conf_read_file',)
command terminated with exit code 1

###file or host  where  am running kubectl
$ cat /var/lib/rook/rook-ceph/rook-ceph.config
[global]
fsid                      = 0905f0ba-b9b2-4550-adcb-244f05b9a0d1
run dir                   = /var/lib/rook/rook-ceph
mon initial members       = a b c
mon host                  = 10.96.96.18:6789,10.106.223.0:6789,10.103.185.118:6789
log file                  = /dev/stderr
mon cluster log file      = /dev/stderr
public addr               = 172.17.0.10
cluster addr              = 172.17.0.10
mon keyvaluedb            = rocksdb
mon_allow_pool_delete     = true
mon_max_pg_per_osd        = 1000
debug default             = 0
debug rados               = 0
debug mon                 = 0
debug osd                 = 0
debug bluestore           = 0
debug filestore           = 0
debug journal             = 0
debug leveldb             = 0
filestore_omap_backend    = rocksdb
osd pg bits               = 11
osd pgp bits              = 11
osd pool default size     = 1
osd pool default min size = 1
osd pool default pg num   = 100
osd pool default pgp num  = 100
crush location            = root=default host=minikube
rbd_default_features      = 3
fatal signal handlers     = false

[client.admin]
keyring = /var/lib/rook/rook-ceph/client.admin.keyring
```

am not sure how this works, do `/var/lib/rook/rook-ceph/rook-ceph.config` shared from the host with mon pods?

Comment (leseb) [reply Documentation/ceph-csi-drivers.md]: You have to run this command from either the operator or a toolbox. A mon pod does not have any ceph.conf that has the mon address, everything is passed on the CLI.

**Comment by Madhu-1** [comment]
@ktdreyer @travisn  added example command PTAL

**Comment by Madhu-1** [comment]
resolved merge conflict

**Pull request merged**